### ISSUE TITLE
Python 2.7 support

### ISSUE THREAD

**Issue by datakurre** [GordonSo/scimschema#1]
I need to implement SCIM provider support for a Python 2.7 based application and would like to re-use this fine library for validating incoming requests.

I have a Python 2 compatible version already at https://github.com/GordonSo/scimschema/compare/master...datakurre:2b3420e9a0de54628df2eaa1d8472af1d45eaa82 but because Python 2 is becoming obsolete, the question is, should I provide a pull request for you to maintain it or just for the project?

(Yes, I know we should port our application to Python 3, but we need to have SCIM support  on the Python 2 version already before we can port.)

**Comment by GordonSo** [comment]
Hey @datakurre feel free to make the PR, I will merge as soon as the tests are shown to be green.

**Comment by datakurre** [comment]
Thanks! I’ll do the pull within two weeks. I already made in-house releases and I have to clean up testing/packaging stuff to make acceptable pull.

I hope that you’ll also accept black-formatted code-base :)

Test do pass for both py2 and py3. Only hard part was the test comparing error messages. There I refactored the messages to show JSON-formatted snippets to avoid differences between py2 unicode string representations and py3 strings. That should also
make error messages better,
because they are shown to users making JSON requests anyway :)
